pymc/aesaraf.py

```diff
@@ -198,22 +170,40 @@ def extract_obs_data(x: TensorVariable) -> np.ndarray:
     raise TypeError(f"Data cannot be extracted from {x}")
 
 
+def extract_rv_and_value_vars(
+    random_var: TensorVariable,
+) -> Tuple[TensorVariable, TensorVariable]:
+    """Return a random variable and it's observations or value variable, or ``None``.
+
+    Parameters
+    ==========
+    var
+        A variable corresponding to a ``RandomVariable``.
+
+    Returns
+    =======
+    The first value in the tuple is the ``RandomVariable``, and the second is the
+    measure/log-likelihood value variable that corresponds with the latter.
+
+    """
+    if not random_var.owner:
+        return None, None
```

Comment:
It looks like this isn’t tested. Why should it return `None` instead of raising something like a `TypeError`?

Reply:
I think this is for input variables that don't have an `owner` like `at.scalar()`